### PR TITLE
users: General cleanups

### PR DESCRIPTION
**Pull request by skobyda**
Split from https://github.com/cockpit-project/cockpit/pull/18234

### CODE REVIEW

**Review by martinpitt — APPROVED** [review]
Looks sensible enough, thanks!

File: pkg/users/account-create-dialog.js [hunk 1]
```diff
@@ -170,19 +172,42 @@ function suggest_username(realname) {
...
+        shell: null,
     };
     let errors = { };
 
     let old_password = null;
     let user_name_dirty = false;
 
+    function get_defaults() {
+        return cockpit.spawn(["useradd", "-D"], { superuser: "require" })
+                .catch(() => console.warn("Could not get useradd defaults. 'useradd -D' failed"))
```
Comment (martinpitt) [inline pkg/users/account-create-dialog.js:193]: Please add `err: "message"` option and show the exception in the log, to make this debuggable.

File: pkg/users/account-create-dialog.js [hunk 2]
```diff
@@ -167,22 +169,45 @@ function suggest_username(realname) {
...
+        shell: null,
     };
     let errors = { };
 
     let old_password = null;
     let user_name_dirty = false;
 
+    function get_defaults() {
+        return cockpit.spawn(["useradd", "-D"], { superuser: "require", err: "message" })
+                .catch(e => console.warn("Could not get useradd defaults: ", e.message))
```
Comment (cockpituous) [inline pkg/users/account-create-dialog.js:193]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18471-20230314-135423-f2974365-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by skobyda** [comment]
A lot of test coverage issues here, but that's because a lot of the code from here will be tested at https://github.com/cockpit-project/cockpit/pull/18234/, since this is mostly a preparation PR

File: pkg/users/account-create-dialog.js [hunk 3]
```diff
@@ -167,22 +169,44 @@ function suggest_username(realname) {
...
```
Comment (cockpituous) [inline pkg/users/account-create-dialog.js:193]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18471-20230316-000158-3ca534fe-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/account-create-dialog.js [hunk 4]
```diff
@@ -167,22 +169,44 @@
...
+                .then(defaults => {
+                    let shell = null;
+                    let base_home_dir = null;
+                    defaults.split("\n").forEach(item => {
+                        if (item.indexOf("SHELL=") === 0) {
+                            shell = item.split("=")[1] || "/bin/bash";
+                        } else if (item.indexOf("HOME=") === 0) {
+                            base_home_dir = item.split("=")[1] || "";
```
Comment (cockpituous) [inline pkg/users/account-create-dialog.js:201]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18471-20230316-000158-3ca534fe-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/parsers.js [hunk 5]
```diff
@@ -77,3 +77,19 @@ function parse_group_content(content) {
 export const etc_group_syntax = {
     parse: parse_group_content
 };
+
+function parse_shells_content(content) {
+    content = (content || "").trim();
+    if (!content) {
+        console.warn("Couldn't read /etc/shells");
+        return [];
```
Comment (cockpituous) [inline pkg/users/parsers.js:85]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18471-20230316-000158-3ca534fe-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by martinpitt — APPROVED** [review]
Thanks!

**Pull request merged**